doc/doxygen/man/man3/Contadors.3
.TH "Contadors" 3 "Dl Jun 1 2020" "Version v3" "Escacs" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Contadors \- Gestiona els dos contadors per a cada color\&. Innacció i \fBEscacs\fP seguits\&.  

.SH SYNOPSIS
.br
.PP
.SS "Mètodes públics"

.in +1c
.ti -1c
.RI "\fBContadors\fP ()"
.br
.ti -1c
.RI "\fBContadors\fP (int \fBcontadorInaccioBlanques\fP, int \fBcontadorInaccioNegres\fP, int \fBcontadorEscacBlanques\fP, int \fBcontadorEscacNegres\fP)"
.br
.ti -1c
.RI "void \fBmodificarContadorEscac\fP (int val, \fBColor\fP col)"
.br
.ti -1c
.RI "void \fBmodificarContadorInaccio\fP (int val, \fBColor\fP col)"
.br
.ti -1c
.RI "int \fBobtenirContadorEscac\fP (\fBColor\fP col)"
.br
.ti -1c
.RI "int \fBobtenirContadorInaccio\fP (\fBColor\fP col)"
.br
.ti -1c
.RI "void \fBposarAZeroContadorEscac\fP (\fBColor\fP col)"
.br
.ti -1c
.RI "void \fBposarAZeroContadorInaccio\fP (\fBColor\fP col)"
.br
.ti -1c
.RI "\fBContadors\fP \fBdeepCopy\fP ()"
.br
.in -1c
.SS "Atributs Privats"

.in +1c
.ti -1c
.RI "int \fBcontadorInaccioBlanques\fP"
.br
.ti -1c
.RI "int \fBcontadorInaccioNegres\fP"
.br
.ti -1c
.RI "int \fBcontadorEscacBlanques\fP"
.br
.ti -1c
.RI "int \fBcontadorEscacNegres\fP"
.br
.in -1c
.SH "Descripció Detallada"
.PP 
Gestiona els dos contadors per a cada color\&. Innacció i \fBEscacs\fP seguits\&. 


.PP
\fBAutor\fP
.RS 4
Eloi Quintana 
.PP
u1963141 
.RE
.PP

.SH "Documentació del Constructor i el Destructor"
.PP 
.SS "Contadors\&.Contadors ()"

.PP
\fBPrecondició\fP
.RS 4
- @Post S'han assignat els valors dels contadors ls valor 0; 
.RE
.PP

.SS "Contadors\&.Contadors (int contadorInaccioBlanques, int contadorInaccioNegres, int contadorEscacBlanques, int contadorEscacNegres)"

.PP
\fBPrecondició\fP
.RS 4
- @Post S'han assignat els valors dels contadors als valors dels parametres; 
.RE
.PP

.SH "Documentació de les Funcions Membre"
.PP 
.SS "\fBContadors\fP Contadors\&.deepCopy ()"

.PP
\fBPrecondició\fP
.RS 4
- @Post s'ha realitzat una còpia de valors de l'objecte 
.RE
.PP
\fBRetorna\fP
.RS 4
retorna un nou objecte amb copia realitzada\&. 
.RE
.PP

.SS "void Contadors\&.modificarContadorEscac (int val, \fBColor\fP col)"

.PP
\fBPrecondició\fP
.RS 4
El color es Blanques o Negres @Post S'ha sumat/restat el valor d'entrada de Escac 
.RE
.PP
\fBParàmetres\fP
.RS 4
\fIval\fP Valor en la operació\&. 
.br
\fIcol\fP \fBColor\fP on s'aplicarà l'operació 
.RE
.PP

.SS "void Contadors\&.modificarContadorInaccio (int val, \fBColor\fP col)"

.PP
\fBPrecondició\fP
.RS 4
El color es Blanques o Negres @Post S'ha sumat/restat el valor d'entrada de Inacció 
.RE
.PP
\fBParàmetres\fP
.RS 4
\fIval\fP Valor en la operació\&. 
.br
\fIcol\fP \fBColor\fP on s'aplicarà l'operació 
.RE
.PP

.SS "int Contadors\&.obtenirContadorEscac (\fBColor\fP col)"

.PP
\fBPrecondició\fP
.RS 4
El color es Blanques o Negres @Post s'ha retornat el valor del contador de Escac del color donat\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
enter del contador seleccionat\&. 
.RE
.PP
\fBParàmetres\fP
.RS 4
\fIcol\fP \fBColor\fP on s'aplicarà l'operació 
.RE
.PP

.SS "int Contadors\&.obtenirContadorInaccio (\fBColor\fP col)"

.PP
\fBPrecondició\fP
.RS 4
El color es Blanques o Negres @Post s'ha retornat el valor del contador d'inaccio del color donat\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
enter del contador seleccionat\&. 
.RE
.PP
\fBParàmetres\fP
.RS 4
\fIcol\fP \fBColor\fP on s'aplicarà l'operació 
.RE
.PP

.SS "void Contadors\&.posarAZeroContadorEscac (\fBColor\fP col)"

.PP
\fBPrecondició\fP
.RS 4
El color es Blanques o Negres @Post s'ha posat a valor 0 el contador d'escac del color donat\&. 
.RE
.PP
\fBParàmetres\fP
.RS 4
\fIcol\fP \fBColor\fP on s'aplicarà l'operació 
.RE
.PP

.SS "void Contadors\&.posarAZeroContadorInaccio (\fBColor\fP col)"

.PP
\fBPrecondició\fP
.RS 4
El color es Blanques o Negres @Post s'ha posat a valor 0 el contador d'inaccio del color donat\&. 
.RE
.PP
\fBParàmetres\fP
.RS 4
\fIcol\fP \fBColor\fP on s'aplicarà l'operació 
.RE
.PP

.SH "Documentació de les Dades Membre"
.PP 
.SS "int Contadors\&.contadorEscacBlanques\fC [private]\fP"
Contador de \fBEscacs\fP de \fBJugador\fP Blanc 
.SS "int Contadors\&.contadorEscacNegres\fC [private]\fP"
Contador de Enrocs de \fBJugador\fP Negre 
.SS "int Contadors\&.contadorInaccioBlanques\fC [private]\fP"
Contador de Inaccio de \fBJugador\fP Blanc 
.SS "int Contadors\&.contadorInaccioNegres\fC [private]\fP"
Contador de Inaccio de \fBJugador\fP Negre 

.SH "Autor"
.PP 
Generat automàticament per Doxygen per a Escacs a partir del codi font\&.


src/Contadors.java
/**
 *  @file Contadors.java
 *  @brief Classe Contadors
 */

/** 
 *  @class Contadors
 *  @brief Gestiona els dos contadors per a cada color. Innacció i Escacs seguits.
 *  @author Eloi Quintana
 *  @author u1963141
 */

public class Contadors {
    
    /**
     * Contador de Inaccio de Jugador Blanc
     */
    private int contadorInaccioBlanques;

    /**
     * Contador de Inaccio de Jugador Negre
     */
    private int contadorInaccioNegres;

    /**
     * Contador de Escacs de Jugador Blanc
     */
    private int contadorEscacBlanques;

    /**
     * Contador de Enrocs de Jugador Negre
     */
    private int contadorEscacNegres;
    
    
    /**
    * @pre -
    * @Post S'han assignat els valors dels contadors ls valor 0;
    */
    public Contadors(){
        this.contadorEscacBlanques   = 0;
        this.contadorEscacNegres     = 0;
        this.contadorInaccioBlanques = 0;
        this.contadorInaccioNegres   = 0;
    }
    
    /**
    * @pre -
    * @Post S'han assignat els valors dels contadors als valors dels parametres;
    */
    public Contadors(int contadorInaccioBlanques, int contadorInaccioNegres, int contadorEscacBlanques, int contadorEscacNegres){
        this.contadorEscacBlanques   = contadorEscacBlanques;
        this.contadorEscacNegres     = contadorEscacNegres;
        this.contadorInaccioBlanques = contadorInaccioBlanques;
        this.contadorInaccioNegres   = contadorInaccioNegres;
    }
    
    /**
    * @pre El color es Blanques o Negres
    * @Post S'ha sumat/restat el valor d'entrada de Escac
    * @param val Valor en la operació.
    * @param col Color on s'aplicarà l'operació
    */
    public void modificarContadorEscac(int val, Color col){
        if(col == Color.Blanques)
            contadorEscacBlanques += val;
        else
            contadorEscacNegres += val;   
    }
    
    /**
    * @pre El color es Blanques o Negres
    * @Post S'ha sumat/restat el valor d'entrada de Inacció
    * @param val Valor en la operació.
    * @param col Color on s'aplicarà l'operació
    */
    public void modificarContadorInaccio(int val, Color col){
        if(col == Color.Blanques)
            contadorInaccioBlanques += val;
        else
            contadorInaccioNegres += val;
    }
    
    /**
    * @pre El color es Blanques o Negres
    * @Post s'ha retornat el valor del contador de Escac del color donat.
    * @return enter del contador seleccionat.
    * @param col Color on s'aplicarà l'operació
    */
    public int obtenirContadorEscac(Color col){
        if(col == Color.Blanques)
            return contadorEscacBlanques;
        else
            return contadorEscacNegres;
    }
    
    /**
    * @pre El color es Blanques o Negres
    * @Post s'ha retornat el valor del contador d'inaccio del color donat.
    * @return enter del contador seleccionat.
    * @param col Color on s'aplicarà l'operació
    */
    public int obtenirContadorInaccio(Color col){
        if(col == Color.Blanques)
            return contadorInaccioBlanques;
        else
            return contadorInaccioNegres;
    }
    
    /**
    * @pre El color es Blanques o Negres
    * @Post s'ha posat a valor 0 el contador d'escac del color donat.
    * @param col Color on s'aplicarà l'operació
    */
    public void posarAZeroContadorEscac(Color col){
        if(col == Color.Blanques)
            contadorEscacBlanques = 0;
        else
            contadorEscacNegres = 0;
    }
    
    /**
    * @pre El color es Blanques o Negres
    * @Post s'ha posat a valor 0 el contador d'inaccio del color donat.
    * @param col Color on s'aplicarà l'operació
    */
    public void posarAZeroContadorInaccio(Color col){
        if(col == Color.Blanques)
            contadorInaccioBlanques = 0;
        else
            contadorInaccioNegres = 0;
    }
    
    /**
    * @pre -
    * @Post s'ha realitzat una còpia de valors de l'objecte
    * @return retorna un nou objecte amb copia realitzada.
    */
    public Contadors deepCopy(){
        return new Contadors(contadorInaccioBlanques, contadorInaccioNegres, contadorEscacBlanques, contadorEscacNegres);
    }
    
}
